browser/js/product/products.controller.js

```diff
@@ -29,15 +29,61 @@ app.controller('ProductsController', function ($scope, products) {
 
 	$scope.$watchCollection('categoryModel', function () {
 		$scope.categoryResults = [];
-	    angular.forEach($scope.categoryModel, function (value, key) {
-	      if (value) {
-			$scope.categoryResults.push(key);
-	      }
-	    });
-	  });
-
+		angular.forEach($scope.categoryModel, function (value, key) {
+			if (value) {
+				$scope.categoryResults.push(key);
+			}
+		});
 	});
 
-app.controller('ProductController', function ($scope, product) {
+	$scope.addToCart = function (productId){
+		// add 1 of item to cart.
+		// CartFactory.addToCart(productId, 1);	
+	};
+});
+
+app.controller('ProductController', function ($scope, product, ReviewFactory, Session) {
 	$scope.product = product;
+	$scope.item = {quantity:1};
+
+	$scope.addToCart = function (){
+		if ($scope.item.quantity<1) $scope.item.quantity=1;
+		$scope.item.product = $scope.product._id;
+
+		console.log("Item", $scope.item);
+		// $scope.item = {quantity: 1, product: "56d6211a41380ba6524bf3e9"}
+
+		// CartFactory.addToCart($scope.item.product, $scope.item.quantity);
+	};
+
+	$scope.areThereProductReviews = function () {
+		return $scope.product.reviews.length>0;
+	};
+
+	$scope.quantitySelected = function (){
+		return $scope.item.quantity>1;
```

Comment:
Just a thought -- we could make this name more like 'isQuantitySelected' because it seems to return a boolean?